course4/homework-string/test.cc

```diff
@@ -120,16 +121,31 @@ void test2()
 	assert(string4.length() == 0);
 }
 
+template<typename StringType>
+void iteratorTest()
+{
+  //const iteration test
+  const StringType s("ABCBB");
+  int b_num = std::count_if(s.begin(), s.end(), [](const char& c){return c=='B';});
```

Comment:
the purpose of passing params by const ref is to avoid copying of _expensive to copy_ object. On case of simple types, like ```char``` (aka 8-bit integer) it has no point, as the copying is cheap.